### PR TITLE
[RFR] Add phar, and automatic update

### PR DESCRIPTION
**Pull request by blazarecki**
This PR is in WIP stage.

The idea is to automatically build `docker-hostmanager.phar` and push it in `gh-page`.

There are some thing to do in order to make it work:
- [x] Travis integration
- [x] Add simple test
- [x] Box support for building phar
- [x] Add console components
- [x] Dump new version to github releases
- [x] Automate all the things (with travis)

You can see travis build here : https://travis-ci.org/blazarecki/docker-hostmanager/

EDIT:

You can see travis build here : https://travis-ci.org/blazarecki/docker-hostmanager/branches
You can see the release here : https://github.com/blazarecki/docker-hostmanager/releases

If you want to deploy a new phar follow the following step :
1. Tag a commit on master : `git tag -a "1.0"`
2. Push tag and commits : `git push --follow-tags`

### CODE REVIEW

**Comment by blazarecki** [comment]
What do thing of this PR @iamluc ?

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /vendor/
+/bin
```
Comment (iamluc) [inline .gitignore]: IMO, we should ignore only the phar

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+before_deploy:
+    - curl -LSs http://box-project.github.io/box2/installer.php | php
+    - php box.phar build
+
+deploy:
+  provider: releases
+  skip_cleanup: true
+  api_key:
+    secure: NuHZx1ORZGojbPuY2ZZrB1BNfe2UUmO5wBv4dM6Qu2WluU0zp96LmKlOsbNml6k167cKj4R1AHgbVsbBARjtUeoYNBYti1G5/m2iXjRRqfPGNkuhG7A2THnYtrLqRb2jCI3hypsZSpOtj//4N+Q3CEKT9nd69HeV90/cAFd+YgDN5ldk9mhHbMnFAJ5hAFzYzUg2zukExVWObWkQ9Y3nP/zJ8jQvmvfiS2yLO+DjPEDH63rQorCbFh9SI0hSWXUQxTtbUk8Jj1H46Mrp9RAZlNhfEL2HiUt4S6lKciCQxkZ3P9yXp8ebjYPqH7fPPhn8G3ewYQqGFG/AYQGX463eyh0VAIt4PbJyIeqtEoPKC1h16b7a9tVt2kroQMImkYV1pwKg/aNZ0x9T3PXBjaCU8Mb5qrUvYpIoxjONI2qxi6lDV0kikumzhFnbfayFTA004ybl+yL1Ag3LxXyqo2jeu3vzX0e9kSDbQD1Cf4lp0riDkbKIDrJ8Ptqv67QV+dcAPyynDkRYDvPoSRKnraVKtHeRu1P5XHH7gpL/xdU8UnvO83ofDaLu923fkDE/dlGNhR/MU8P4EMWC4U/y0ROp1Wx27+qatrzCFZBlJVjtq6urTZ5Z6fMrD9f/Nelu8UR4CI8hnWMFl+UQVoQpSDYYVh71DHPU1QK+A/TpGI4GYm8=
```
Comment (iamluc) [inline .travis.yml:22]: euh.... are you sure this should be public ?

File: bin/docker-hostmanager [hunk 3]
```diff
@@ -1,15 +1,11 @@
 #!/usr/bin/env php
 <?php
 
-if (PHP_SAPI !== 'cli') {
```
Comment (iamluc) [inline bin/docker-hostmanager:4]: This test should be keep

File: composer.json [hunk 4]
```diff
@@ -14,6 +18,6 @@
         }
     ],
     "autoload": {
-        "psr-4": {"DockerHostManager\\": "src/"}
```
Comment (iamluc) [inline composer.json:17]: IMO, directory "DockerHostManager" is not needed in FS. I prefer let the autoload like it is actually

File: src/DockerHostManager/Command/RunCommand.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                InputArgument::OPTIONAL,
+                'The TLD to use',
+                getenv('TLD') ?: '.docker'
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $app = new Application(
```
Comment (iamluc) [inline src/DockerHostManager/Command/RunCommand.php]: The class `Application` is not needed anymore. You can move the code in this command.

File: src/DockerHostManager/Command/RunCommand.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RunCommand extends Command
+{
+    protected function configure()
+    {
+        $this
+            ->setName('run')
```
Comment (iamluc) [inline src/DockerHostManager/Command/RunCommand.php]: I suggest to change the name: `synchronize-hosts`, `listen`, `update`, ... ??

File: .travis.yml (hunk 2 above)
Comment (blazarecki) [inline .travis.yml:22]: Yes, this is an encrypted var, see https://docs.travis-ci.com/user/encryption-keys/, By the way, once this PR is merged, you'll need to change this api_key to one of your, if you want to be able to deploy releases.

File: bin/docker-hostmanager (hunk 3 above)
Comment (blazarecki) [inline bin/docker-hostmanager:4]: ok

File: .gitignore (hunk 1 above)
Comment (blazarecki) [inline .gitignore]: ok

File: .travis.yml (hunk 2 above)
Comment (iamluc) [inline .travis.yml:22]: nice feature !

File: README.md [hunk 7]
```diff
@@ -21,10 +21,24 @@ The easiest way is to use the docker image
...
+
+If you don't want to use the docker image you can download a PHAR executable here : https://github.com/iamluc/docker-hostmanager/releases
+
+And then run it:
+
+```console
+$ php docker-hostmanager.phar
+```
+
 ### TIPS
```
Comment (iamluc) [inline README.md]: The tips should stay in the docker section

**Comment by iamluc** [comment]
@blazarecki Will you have time to finish this ?

**Comment by blazarecki** [comment]
Yes, I'm on it

File: composer.json (hunk 4 above)
Comment (blazarecki) [inline composer.json:17]: Done

File: src/DockerHostManager/Command/RunCommand.php (hunk 5 above)
Comment (blazarecki) [inline src/DockerHostManager/Command/RunCommand.php]: Are you sure ? I think it's better in a separate class. What do you think ?

File: src/DockerHostManager/Command/RunCommand.php (hunk 6 above)
Comment (blazarecki) [inline src/DockerHostManager/Command/RunCommand.php]: Ok I choose `synchronize-hosts`.

File: README.md (hunk 7 above)
Comment (blazarecki) [inline README.md]: Ok, my bad

**Comment by blazarecki** [comment]
I update the PR and rebase it.

**Comment by iamluc** [comment]
LGTM, but can you rebase on develop and change PR target ?

**Pull request closed without merging**